specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,974 @@
+  cancelling,
+}
+
+@doc("Base task object.")
+model TaskIdentifier {
+  @doc("task name")
+  taskName?: string;
+}
+
+@clientName("TaskContent", "csharp")
+@doc("Base parameters object for a text analysis task.")
+model TaskParameters {
+  @doc("logging opt out")
+  loggingOptOut?: boolean = false;
+}
+
+@clientName("PreBuiltTaskContent", "csharp")
+@doc("Parameters object for a text analysis task using pre-built models.")
+model PreBuiltTaskParameters is TaskParameters {
+  @doc("model version")
+  modelVersion?: string = "latest";
+}
+
+alias ShowStatsQueryParameter = QueryParametersTrait<{
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+}>;
+
+alias AnalyzeTextJobQueryParameters = QueryParametersTrait<{
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+
+  @doc("The maximum number of resources to return from the collection.")
+  @query
+  top?: int32;
+
+  @doc("An offset into the collection of the first resource to be returned.")
+  @query
+  skip?: int32;
+}>;
+
+@doc("Query parameters for the AnalyzeText operation.")
+model AnalyzeTextQueryParameters {
+  ...Azure.Core.Foundations.ApiVersionParameter;
+
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+}
+
+@doc("job state")
+model JobState {
+  @doc("display name")
+  displayName?: string;
+
+  @doc("Date and time job created")
+  createdDateTime: utcDateTime;
+
+  @doc("Date and time job expires")
+  expirationDateTime?: utcDateTime;
+
+  @doc("job ID")
+  @key
+  @visibility("read")
+  jobId: uuid;
+
+  @doc("last updated date and time")
+  lastUpdatedDateTime: utcDateTime;
+
+  @doc("status")
+  status: State;
+
+  @doc("errors")
+  errors?: Array<Error>;
+
+  @doc("next link")
+  nextLink?: string;
+}
+
+@doc("job errors")
+model JobErrors {
+  @doc("errors")
+  errors?: Array<Error>;
+}
+
+@doc("Contains details of errors encountered during a job execution.")
+model DocumentError {
+  @doc("The ID of the input document.")
+  id: string;
+
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type"
+  @doc("Error encountered.")
+  error: Error;
+}
+
+@doc("Contains details of warnings encountered during a job execution.")
+model InputWarning {
+  @doc("Warning code.")
+  code: string;
+
+  @doc("Warning message.")
+  message: string;
+
+  @doc("A JSON pointer reference indicating the target object.")
+  targetRef?: string;
+}
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("if showStats=true was specified in the request this field will contain information about the request payload.")
+model RequestStatistics is Record<unknown> {
```

Comment:
Why is this a `Record<unknown>`? From what I can recall, there are no additional properties possible here, right?

Reply:
The original openapi spec has this object with the parameter `"additionalProperties": true`.

Reply:
What additional fields will the service send back? This seem like a bug in the previous description of the service.

Reply:
The `Record<unknown>` should be removed. As there is nothing else sent from the api

Reply:
removed.